src/backend/expungeservice/models/charge_types/charge.py

```diff
@@ -34,7 +39,7 @@ def convicted(self):
 
     def recent_conviction(self):
         ten_years_ago = (date_class.today() + relativedelta(years=-10))
-        if self.convicted():
+        if self.convicted() and self.disposition:
```

Comment:
This extra check of `self.disposition` is not needed. Likewise the self.disposition on line 38 is also not needed.